aiohttp/http_writer.py

```diff
@@ -138,29 +139,39 @@ def __init__(self, stream, loop, acquire=True):
         self.output_size = 0
 
         self._eof = False
-        self._buffer = []
         self._compress = None
         self._drain_waiter = None
 
         if self._stream.available:
             self._transport = self._stream.transport
+            self._buffer = None
             self._stream.available = False
         elif acquire:
             self._stream.acquire(self)
 
     def set_transport(self, transport):
         self._transport = transport
 
-        chunk = b''.join(self._buffer)
-        if chunk:
-            transport.write(chunk)
-            self._buffer.clear()
+        if self._buffer is not None:
+            for b in self._buffer:
+                transport.write(b)
+            self._buffer = None
 
         if self._drain_waiter is not None:
             waiter, self._drain_waiter = self._drain_waiter, None
             if not waiter.done():
                 waiter.set_result(None)
 
+    @asyncio.coroutine
+    def get_transport(self):
+        if self._transport is None:
+            if self._drain_waiter is None:
+                self._drain_waiter = create_future(self.loop)
+            yield from self._drain_waiter
+
+        assert self._transport is not None
```

Comment:
nothing that i can think of, i was using the asserts as safeguards/way to comment the code. I'll remove them